sdk/go/examples_test.go

```diff
@@ -210,10 +210,9 @@ func ExampleHost_Workdir() {
 		panic(err)
 	}
 
-	lines := strings.Split(strings.TrimSpace(readme), "\n")
```

Comment:
@vito Wasn't sure what to do here.

This test is using `dagger.Connect(ctx, dagger.WithWorkspace("."))`.

Locally, that's the `sdk/go` directory. However when executed in a container, it will use the "parent"'s workdir (which is the root repo).

I was trying to find another way to make this test work that could run from either directory (either this, or remove the `// Output` at the end.

Reply:
I think we should just make this example self-contained. It's probably too implicit as it is. Something like:

```go
dir, err := os.MkdirTemp("", "example*")
if err != nil {
  panic(err)
}

defer os.RemoveAll(dir)

err = os.WriteFile(filepath.Join(dir, "foo"), []byte("hello"), 0644)
if err != nil {
  panic(err)
}

client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))
// ...
```

It's more verbose but way more obvious what it'll do.

Reply:
> client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))

The trick here is `WithWorkdir()` is a no op with dagger-in-dagger because it's only taken into account at engine.Start which already happened

Reply:
How about now? A bit hackish, but it's self contained

Reply:
Err -- I hit undo.

I was trying to write a "hello.txt" file and read it back -- however since there's no more "workdir write", os.WriteFile() and Workdir() didn't match

Reply:
OK with this for now, hopefully we can clean it up/make it clearer once we support remote engine.